frontend/src/metabase/lib/entities.js

```diff
@@ -259,14 +259,12 @@ export function createEntity(def) {
         );
 
         if (notify) {
-          // pick only the attributes that were updated
-          const undoObject = _.pick(
-            originalObject,
-            ...Object.keys(updatedObject || {}),
-          );
-          // https://github.com/metabase/metabase/pull/20874#pullrequestreview-902384264
-          const canUndo = !hasCircularReference(undoObject);
-          if (notify.undo && canUndo) {
+          if (notify.undo) {
```

Comment:
Revert the code to check circular reference from https://github.com/metabase/metabase/pull/20874 as we don't need to undo changes on the data model page anymore https://github.com/metabase/metabase/pull/20915#pullrequestreview-904419004